head/firmware/motor_hardware.c

```diff
@@ -356,11 +375,32 @@ void HAL_TIM_Base_MspInit(TIM_HandleTypeDef *htim) {
     }
 }
 
+uint16_t encoder_pulse_count(TIM_HandleTypeDef *htim){
+    if (htim == &htim2){
+        enc_pulse_counter = __HAL_TIM_GET_COUNTER(&htim2);
+    }
+    else if(htim == &htim3){
+        enc_pulse_counter = __HAL_TIM_GET_COUNTER(&htim3);
+    }
+    return enc_pulse_counter;
+}
+
+void reset_encoder_counter(TIM_HandleTypeDef *htim){
```

Comment:
I don't see where this function is being called.

Reply:
it's not being called anywhere - this is a first implementation intended to work basically only under a debugger, and make sure we got the hardware handling stuff right. it may get discarded though, we might want to call it as a dummy thing from somewhere